File: client/local_client.py
from game.client import Client
from game.tic_tac_toe_state import TicTacToeState


class LocalClient(Client):
    def __init__(self) -> None:
        self._queue: list[TicTacToeState] = []
        self._next_index = 0

    def send_state(self, state: TicTacToeState) -> None:
        self._queue.append(state)

    def receive_state(self) -> TicTacToeState | None:
        if self._next_index < len(self._queue):
            self._next_index += 1
            return self._queue[self._next_index - 1]
        return None

    def close(self) -> None:
        pass

File: local/localClient.go
package local

import "tic_tac_toe/core"

type LocalClient struct {
	queue     []core.GameState
	nextIndex int
}

func (client *LocalClient) SendState(state core.GameState) {
	client.queue = append(client.queue, state)
}

func (client *LocalClient) ReceiveState() *core.GameState {
	if client.nextIndex < len(client.queue) {
		client.nextIndex++
		return &client.queue[client.nextIndex-1]
	}
	return nil
}

func (client *LocalClient) Close() {
}
